logstash-core/lib/logstash/runner.rb

```diff
@@ -148,25 +148,30 @@ def run(args)
     begin
       LogStash::SETTINGS.from_yaml(LogStash::SETTINGS.get("path.settings"))
     rescue => e
-      @logger.subscribe(STDOUT)
       @logger.warn("Logstash has a new settings file which defines start up time settings. This file is typically located in $LS_HOME/config or /etc/logstash. If you installed Logstash through a package and are starting it manually please specify the location to this settings file by passing in \"--path.settings=/path/..\" in the command line options")
       @logger.fatal("Failed to load settings file from \"path.settings\". Aborting...", "path.settings" => LogStash::SETTINGS.get("path.settings"), "exception" => e.class, "message" => e.message)
       exit(-1)
     end
 
+    # initialize logging
+    logging_ctx = org.apache.logging.log4j.LogManager.getContext(false)
+    logging_ctx.setConfigLocation(java.io.File.new(setting("path.settings") + "/log4j.properties").toURI)
+
     super(*[args])
   end
 
   def execute
     require "logstash/util"
     require "logstash/util/java_version"
     require "stud/task"
-    require "cabin" # gem 'cabin'
-    require "logstash/logging/json"
 
     # Configure Logstash logging facility, this need to be done before everything else to
     # make sure the logger has the correct settings and the log level is correctly defined.
-    configure_logging(setting("path.log"), setting("log.level"))
+    LogStash::Logging::Logger::configure_logging(setting("log.level"), setting("path.log"))
```

Comment:
Hhmm, several things here :
- Presently, `configure_logging` method implementation updates log level for one logger (path).
- As confirmed by @ph, to set `path.log` and `log.level`, the right way is to set these settings as system properties using `System.setProperty()` and then reference them in config using `${sys:option}`. And this code has to be set at line 156, before initializing log4j.
- variable `@logger` is set before log4j is initialized by your code (at line 138). This implies log4j is initialized a first time using a default configuration (because no log4j file found). This default configuration will log on console with error root level.
- This implies warn logs will not be displayed (like line 151)